packages/infra/lib/ihe-stack/ihe-gw-construct.ts

```diff
@@ -0,0 +1,307 @@
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as ecr from "aws-cdk-lib/aws-ecr";
+import * as ecs from "aws-cdk-lib/aws-ecs";
+import * as ecs_patterns from "aws-cdk-lib/aws-ecs-patterns";
+import { ApplicationLoadBalancedTaskImageOptions } from "aws-cdk-lib/aws-ecs-patterns";
+import {
+  ApplicationProtocol,
+  ListenerAction,
+  Protocol,
+} from "aws-cdk-lib/aws-elasticloadbalancingv2";
+import { Function as Lambda } from "aws-cdk-lib/aws-lambda";
+import * as r53 from "aws-cdk-lib/aws-route53";
+import * as r53_targets from "aws-cdk-lib/aws-route53-targets";
+import { IBucket } from "aws-cdk-lib/aws-s3";
+import { Construct } from "constructs";
+import { EnvConfig } from "../../config/env-config";
+import { IHEGatewayProps } from "../../config/ihe-gateway-config";
+import { ecrRepoName } from "../ihe-prereq-stack";
+import { getLambdaUrl as getLambdaUrlShared } from "../shared/lambda";
+import { isProd } from "../shared/util";
+import IHEDBConstruct from "./ihe-db-construct";
+
+export interface IHEGatewayConstructProps {
+  mainConfig: EnvConfig;
+  config: IHEGatewayProps;
+  vpc: ec2.IVpc;
+  // TODO 1377 Implement this
+  // alarmThresholds?: IHEGatewayAlarmThresholds;
+  privateZone: r53.IPrivateHostedZone;
+  db: IHEDBConstruct;
+  alarmAction?: SnsAction | undefined;
+  documentQueryLambda: Lambda;
+  documentRetrievalLambda: Lambda;
+  patientDiscoveryLambda: Lambda;
+  medicalDocumentsBucket: IBucket;
+}
+
+const id = "IHEGateway";
+
+const mainPort = 8443;
+const additionalHTTPPorts = [
+  8081, 8082, 8083, 8084, 8085, 8086, 9091, 9092, 8100, 23067, 16850, 16852, 16853,
+];
+
+export default class IHEGatewayConstruct extends Construct {
+  public readonly server: ecs.IFargateService;
+  public readonly serverAddress: string;
+
+  constructor(scope: Construct, props: IHEGatewayConstructProps) {
+    super(scope, `${id}Construct`);
+
+    const {
+      vpc,
+      mainConfig,
+      config,
+      privateZone,
+      db,
+      documentQueryLambda,
+      documentRetrievalLambda,
+      patientDiscoveryLambda,
+      medicalDocumentsBucket,
+    } = props;
+    const dbAddress = db.server.clusterEndpoint.socketAddress;
+    const dbIdentifier = config.rds.dbName;
+
+    const getLambdaUrl = (arn: string) => {
+      return getLambdaUrlShared({ region: mainConfig.region, arn });
+    };
+
+    const secrets: ApplicationLoadBalancedTaskImageOptions["secrets"] = {
+      DATABASE_PASSWORD: ecs.Secret.fromSecretsManager(db.secret),
+    };
+    const environment: ApplicationLoadBalancedTaskImageOptions["environment"] = {
+      DATABASE: `postgres`,
+      DATABASE_URL: `jdbc:postgresql://${dbAddress}/${dbIdentifier}`,
+      DATABASE_USERNAME: config.rds.userName,
+      INBOUND_XCPD_URL: getLambdaUrl(patientDiscoveryLambda.functionArn),
+      INBOUND_XCA38_URL: getLambdaUrl(documentQueryLambda.functionArn),
+      INBOUND_XCA39_URL: getLambdaUrl(documentRetrievalLambda.functionArn),
```

Comment:
we seem to be flip-flopping between using XCPD vs Patient Discovery all over the place (even in the Mirth env var code), so I suggest we just pick one and stick with it

we can just refer to `XCA38` as `DocumentQuery` everywhere, for example, to obfuscate the profile until we need to reference in the specific Mirth implementation

otherwise, it's hard to come up with guidelines on what to use where

Reply:
Yeah, I totally agree. I think I was trying to keep the config the IHE GW uses, but I couldn't find the matching parameter. After some digging, I found they were not matching (`INBOUND_XCA38_URL ` vs `INBOUND.XCA38`), so I might as well update both places to PD/DQ/DR.

Thanks for bringing this up!